stratum/client.cpp

```diff
@@ -229,10 +229,10 @@ bool client_authorize(YAAMP_CLIENT *client, json_value *json_params)
 		}
 	}
 
-	if (!is_base58(client->username)) {
+	/*if (!is_base58(client->username)) {
```

Comment:
I think that this check was added to prevent that stupid or malicious miners to send wrong username that hurted some pools. If using a nickname is mandatory, would it be possible to limit this change ? I'm not sure how.
https://github.com/tpruvot/yiimp/commit/138b0079bb2c78930aad106ce691d3a733dca6f3

I didn't know that BitCash moved to x16r. I noticed cuckoo27 on your repo  https://github.com/WillyTheCat/bitcashyiimppool Do you think that rpc could be adapted for Grin that use variants of cuckoo (cuckatoo and cuckaroo) ? On github, it seems that several persons are motivated to adding Grin into yiimp

Reply:
If you want you can activate this check again. Then mining would only be possible with the regular addresses, not with the nicknames. Should I activate the check so that you can pull it?
We will fork to x16 in two weeks. So currently the cuckoo cycle mining is still active. 
Do you mean the RPC command which I build into BitCash to create the coinbase transaction? 
This is not needed for Cuckoo cycle. This is only needed for the stealth addresses used in BitCash.
To create a transaction with the stealth addresses correctly is a bit tricky. 
The cuckoo cylce part could be used I think, but needs to changed so that it is compatible with other coins.